### PR TITLE
impl: capture the PiperOrigin-RevId from the googleapis commit

### LINKED ISSUE #13402: Capture the latest `PiperOrigin-RevId` value when updating googleapis SHA
To help the internal automation team we should capture the latest `PiperOrigin-RevId` in the commit message for any PR updating the googleapis SHA.

Most likely this would involve using the commit message from here:

https://github.com/googleapis/google-cloud-cpp/blob/a3368d70df113d1dde5143990b19ecf8f81e21a3/external/googleapis/renovate.sh#L30-L31

And then adding (an optimized version of):

```sh
curl -fsSL \
      -H "Accept: application/vnd.github+json" \
      https://api.github.com/repos/googleapis/googleapis/commits?sha=${COMMIT}  | \
  jq 'map(.commit.message)' | grep -Eo 'PiperOrigin-RevId: [0-9]*' | head -1
```

and then inject that line as part of the commit message in:

https://github.com/googleapis/google-cloud-cpp/blob/a3368d70df113d1dde5143990b19ecf8f81e21a3/external/googleapis/renovate.sh#L72

### PR DESCRIPTION
**Pull request by devbww**
To help the internal automation team, copy the PiperOrigin-RevId from the latest googleapis commit into our SHA-update PR message.

Fixes #13402.

- - -
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/13410)

### CODE REVIEW

File: external/googleapis/renovate.sh [hunk 1]
```diff
@@ -29,6 +29,8 @@ REPO="googleapis/googleapis"
 BRANCH="master"
 COMMIT=$(curl -fsSL -H "Accept: application/vnd.github.VERSION.sha" \
   "https://api.github.com/repos/${REPO}/commits/${BRANCH}")
+PIPERORIGIN_REVID=$(gh api "repos/${REPO}/commits/${COMMIT}" |
```
Comment (coryan) [inline external/googleapis/renovate.sh:32]: Note that not all commits include a `PiperOrigin-RevId`, for example:

https://github.com/googleapis/googleapis/commit/a6bbdede82cbd03af41effc59373cdcd25bf9ed4

My suggestion in the bug was to list the last $N$ commits and pick the latest PiperOrigin-RevId value.  That could fail, but seems unlikely.

Comment (devbww) [reply external/googleapis/renovate.sh:32]: I see.  Thanks.  Let me take another look.

Comment (devbww) [reply external/googleapis/renovate.sh]: OK, I added the reverse scan.

At this stage though, I don't grok whether this satisfies the requirements.  So, we can either plow ahead and see, or revert and regroup to try another day.

**Review by coryan — APPROVED** [review]
I say we plow ahead.  My estimation is that the automation team will need to adapt their code to how the C++ libraries work, whatever we do otherwise.

**Pull request merged**